### ISSUE TITLE
EnC analyzer persists semantic diagnostics

### ISSUE THREAD

**Issue by isadorasophia** [dotnet/roslyn#53158]
1. Debug the following console app:
```csharp
static void Main()
{
    Debugger.Break();

    var allNumbers = new int[] { 1, 2, 3 }; // delete this line
    foreach (var num in allNumbers)
    {
        Console.WriteLine(num + " ");
    }
}
```
2. Delete commented line.
3. Press F10;
4. Errors are reported:
![image](https://user-images.githubusercontent.com/7958854/117079839-7251ee80-acf1-11eb-8b26-c55d3e8eb0b3.png)
5. Ctrl+Z or add the line again.
6. Press F10
7. Change is applied, but error persists until the end of debugging session:
![image](https://user-images.githubusercontent.com/7958854/117079897-91508080-acf1-11eb-8156-1a88c59be839.png)

This seems very likely related to #36061.

**Comment by jinujoseph** [comment]
@tmat i believe this was resolved ...can you validate pls

**Comment by mattscheffer** [comment]
@jinujoseph I can repro this bug in Main 31515.61, That being said, EnC still works fine. Deleting the commented line adds 2 errors at the same location. Re-adding removes 1 of the errors. Debugging is able to continue correction after edits are made. The error does persist for the rest of the debugging session but disappears after the session is complete.
